projects/packages/waf/tests/php/unit/test-waf-request.php

```diff
@@ -327,6 +329,23 @@ public function testGetVarsPostWithJson() {
 		unset( $_SERVER['CONTENT_TYPE'] );
 	}
 
+	/**
+	 * Test that the Waf_Request class returns any parameters when HTTP method isn't POST.
+	 */
+	public function testGetVarsPostHttpMethodNotPost() {
+		$_SERVER['CONTENT_TYPE'] = 'application/x-www-form-urlencoded';
+		$request                 = $this->mock_request(
+			array(
+				'body' => 'value',
+			)
+		);
+		$value                   = $request->get_post_vars();
+		$this->assertIsArray( $value );
+		$this->assertContains( array( 'value', '' ), $value );
+
+		unset( $_SERVER['CONTENT_TYPE'] );
```

Comment:
```suggestion
		$_SERVER['CONTENT_TYPE'] = 'application/x-www-form-urlencoded';
		$request                 = $this->mock_request(
			array(
				'body' => (
					http_build_query( array(
						'str' => 'value',
						'arr' => array( 'a', 'b', 'c' ),
						'obj' => (object) array( 'foo' => 'bar' ),
					) )
				)
			)
		);
		$value                   = $request->get_post_vars();
		$this->assertIsArray( $value );
		$this->assertContains( array( 'str', 'value' ), $value );
		$this->assertContains( array( 'arr[0]', 'a' ), $value );
		$this->assertContains( array( 'arr[1]', 'b' ), $value );
		$this->assertContains( array( 'arr[2]', 'c' ), $value );
		$this->assertContains( array( 'obj[foo]', 'bar' ), $value );

		unset( $_SERVER['CONTENT_TYPE'] );
```

Suggestion: update test with more complex body value, using `http_build_query` to encode an array of key/value pairs.